File: app/src/main/java/com/mytaxiapp/ui/base/BaseActivity.java
package com.mytaxiapp.ui.base;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.mytaxiapp.di.component.VehicleComponent;


/**
 * @author Tosin Onikute.
 */

public abstract class BaseActivity extends AppCompatActivity {

    private VehicleComponent component;

    protected abstract void setupActivity(VehicleComponent component, Bundle savedInstanceState);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setupActivity(component, savedInstanceState);
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
    }



}

File: app/src/main/java/com/mytaxiapp/ui/base/BaseActivity.kt
package com.mytaxiapp.ui.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.mytaxiapp.di.component.VehicleComponent


/**
 * @author Tosin Onikute.
 */

abstract class BaseActivity : AppCompatActivity() {

    private val component: VehicleComponent? = null

    protected abstract fun setupActivity(component: VehicleComponent?, savedInstanceState: Bundle?)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupActivity(component, savedInstanceState)
    }

    override fun onStart() {
        super.onStart()
    }

    override fun onResume() {
        super.onResume()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onStop() {
        super.onStop()
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
    }


}
